### PR TITLE
Add method to client for getting daily peak report

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
Adds a new method for getting a daily peak report for the specified ISO on the specified date. Example: `client.get_daily_peak_report(iso="CAISO", market_date="2024-07-11")`

Returns the raw response from the API as a Python dict.

### CODE REVIEW

File: gridstatusio/tests/test_api.py [hunk 1]
```diff
@@ -834,3 +835,18 @@ def test_pagination():
...
+        ("ERCOT", None, datetime.now().strftime("%Y-%m-%d")),
+        ("CAISO", "2024-07-01", "2024-07-01"),
+        ("spp", datetime(2024, 7, 10), "2024-07-10"),
+    ],
+)
+def test_reports_api(iso, market_date, expected_date):
+    resp = client.get_daily_peak_report(iso=iso, market_date=market_date)
+    assert isinstance(resp, dict)
+    assert resp["ISO"] == iso.upper()
+    assert resp["market_date"] == expected_date
```
Comment (thehomebrewnerd) [inline gridstatusio/tests/test_api.py:852]: For this to pass we need an API key to use that has the correct entitlement to access the endpoint. Not sure if it is best to fix that now, or save this for later?

**Review by WillKoehrsen — APPROVED** [review]

File: gridstatusio/gs_client.py [hunk 2]
```diff
@@ -417,6 +422,37 @@ def get_dataset(
 
         return df
 
+    def get_daily_peak_report(
```
Comment (thehomebrewnerd) [inline gridstatusio/gs_client.py:425]: @kmax12 Is this API in line with your thinking or do we need to update?

**Review by kmax12 — APPROVED** [review]
LGTM

**Comment by kmax12** [comment]
can you add to changelog in this PR under a `v.6.3 - Upcoming` header? 

https://github.com/gridstatus/gridstatusio/blob/main/CHANGELOG.md

**Comment by thehomebrewnerd** [comment]
> can you add to changelog in this PR under a `v.6.3 - Upcoming` header?
> 
> https://github.com/gridstatus/gridstatusio/blob/main/CHANGELOG.md

Added: [c7fd578](https://github.com/gridstatus/gridstatusio/pull/47/commits/c7fd578390ddf1f5b413871be87e1e1089b2fe37)

**Review by kmax12 — APPROVED** [review]

**Pull request merged**